### PR TITLE
sonarcloud code smells : Replace this instanceof check and cast with 'instanceof String string'

### PR DESCRIPTION
**Pull request by vlo-rte**
Nothing in release notes.

### CODE REVIEW

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/configuration/webflux/IpAddressAuthorizationManager.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ public Mono<AuthorizationDecision> check(Mono<Authentication> authenticationMono
 
     private User getAuthenticatedUser(Authentication authentication) {
         User userData = null;
-        if (authentication.getPrincipal()  instanceof CurrentUserWithPerimeters){
-            CurrentUserWithPerimeters current = (CurrentUserWithPerimeters) authentication.getPrincipal();
+        if (authentication.getPrincipal()  instanceof CurrentUserWithPerimeters currentUserWithPerimeters){
+            CurrentUserWithPerimeters current = currentUserWithPerimeters;
             userData = current.getUserData();
```
Comment (quinarygio) [inline services/cards-consultation/src/main/java/org/opfab/cards/consultation/configuration/webflux/IpAddressAuthorizationManager.java:67]: Is it necessary to define 'current' variable? Could we just use "userData = currentUserWithPerimeters .getUserData()" ?

**Review by quinarygio — CHANGES_REQUESTED** [review]
Added comments about not needed variable definitions

File: services/cards-consultation/src/main/java/org/opfab/cards/consultation/configuration/webflux/IpAddressAuthorizationManager.java (hunk 1 above)
Comment (vlo-rte) [reply services/cards-consultation/src/main/java/org/opfab/cards/consultation/configuration/webflux/IpAddressAuthorizationManager.java]: You're right. Done

File: tools/spring/spring-oauth2-utilities/src/main/java/org/opfab/springtools/configuration/oauth/WebSecurityChecks.java [hunk 2]
```diff
@@ -56,11 +56,11 @@ public boolean checkUserIpAddress(Authentication authentication) {
...
         //but is String type if there is no authentication (jira : OC-655)
-        if (authentication.getPrincipal()  instanceof String) {
-            user = (String) authentication.getPrincipal();
+        if (authentication.getPrincipal()  instanceof String principal) {
+            user = principal;
         }
-        else if (authentication.getPrincipal()  instanceof CurrentUserWithPerimeters){
-            CurrentUserWithPerimeters current = (CurrentUserWithPerimeters) authentication.getPrincipal();
+        else if (authentication.getPrincipal()  instanceof CurrentUserWithPerimeters currentUserWithPerimeters){
+            CurrentUserWithPerimeters current = currentUserWithPerimeters;
```
[the remark this answers is not in the document]
Comment (vlo-rte) [reply tools/spring/spring-oauth2-utilities/src/main/java/org/opfab/springtools/configuration/oauth/WebSecurityChecks.java]: You're right. Done

**Review by quinarygio — APPROVED** [review]

**Pull request merged**